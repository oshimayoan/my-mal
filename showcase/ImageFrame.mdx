---
name: ImageFrame
---

import {PropsTable, Playground} from 'docz';
import ImageFrame from '../src/core-ui/ImageFrame';

# ImageFrame

## Properties
<PropsTable of={ImageFrame} />

## Basic Usage
<Playground>
  <ImageFrame 
    imageUri="https://myanimelist.cdn-dena.com/r/334x484/images/anime/1173/92110.jpg?s=9055f67e52d09a3f331b686f02efa37e"
    title="Shingeki no Kyojin Season 3"
  />
</Playground>
