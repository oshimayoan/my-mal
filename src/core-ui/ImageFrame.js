// @flow strict

import React, {Fragment} from 'react';

type Props = $Exact<{
  imageUri: string,
  title: string,
  link?: string,
  dataTest?: string,
}>;

export default function ImageFrame(props: Props) {
  let content = (
    <Fragment>
      <img src={props.imageUri} alt={props.title} width="255" height="340" />
      <div
        style={{
          position: 'absolute',
          left: 0,
          right: 0,
          bottom: 0,
          background: 'rgba(0, 0, 0, 0.5)',
          display: 'flex',
          flex: 1,
          justifyContent: 'center',
          alignItems: 'center',
          padding: '20px 5px',
          color: 'white',
          fontWeight: '400',
        }}
      >
        {props.title}
      </div>
    </Fragment>
  );
  return (
    <div
      style={{
        width: 255,
        height: 340,
        margin: 10,
        background: 'red',
        position: 'relative',
      }}
      data-test={props.dataTest}
    >
      {!props.link ? (
        <div>{content}</div>
      ) : (
        <a href={props.link} target="_blank" rel="noopener noreferrer">
          {content}
        </a>
      )}
    </div>
  );
}
