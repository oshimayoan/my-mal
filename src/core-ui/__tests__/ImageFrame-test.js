// @flow strict-local

import React from 'react';
import {render} from 'react-testing-library';
import ImageFrame from '../ImageFrame';

describe('ImageFrame', () => {
  it('should render without anchor', () => {
    let {container} = render(<ImageFrame imageUri="imageUri" title="Title" />);
    expect(container).toMatchSnapshot();
  });

  it('should render with anchor', () => {
    let {getByTestId} = render(
      <ImageFrame imageUri="imageUri" title="Title" link="link" />,
    );
    let component = getByTestId('imageAnchor');
    expect(component).toMatchSnapshot();
  });
});
