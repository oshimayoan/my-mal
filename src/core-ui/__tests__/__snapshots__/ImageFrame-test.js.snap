// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageFrame should render with anchor 1`] = `
<a
  data-testid="imageAnchor"
  href="link"
  rel="noopener noreferrer"
  target="_blank"
>
  <img
    alt="Title"
    height="340"
    src="imageUri"
    width="255"
  />
  <div
    style="position: absolute; left: 0px; right: 0px; bottom: 0px; background: rgba(0, 0, 0, 0.5); display: flex; justify-content: center; align-items: center; padding: 20px 5px; color: white; font-weight: 400;"
  >
    Title
  </div>
</a>
`;

exports[`ImageFrame should render without anchor 1`] = `
<div>
  <div
    style="width: 255px; height: 340px; margin: 10px; background: red; position: relative;"
  >
    <div>
      <img
        alt="Title"
        height="340"
        src="imageUri"
        width="255"
      />
      <div
        style="position: absolute; left: 0px; right: 0px; bottom: 0px; background: rgba(0, 0, 0, 0.5); display: flex; justify-content: center; align-items: center; padding: 20px 5px; color: white; font-weight: 400;"
      >
        Title
      </div>
    </div>
  </div>
</div>
`;
